src/libcore/slice/mod.rs

```diff
@@ -5956,10 +5956,18 @@ where
             return false;
         }
 
+        #[cfg(bootstrap)]
         if self.as_ptr() == other.as_ptr() {
             return true;
         }
 
+        // While performance would suffer if `guaranteed_eq` just returned `false`
+        // for all arguments, correctness and return value of this function are not affected.
+        #[cfg(not(bootstrap))]
```

Comment:
Why was this changed? This function is not const.

Reply:
Not yet, we're working towards that. I have a local branch where I'm trying to run this in `-Zunleash-the-miri-inside-of-you` mode.